frontend/js/src/settings/Settings.tsx

```diff
@@ -3,25 +3,49 @@ import * as React from "react";
 import { Link } from "react-router-dom";
 import { toast } from "react-toastify";
 import { Helmet } from "react-helmet";
+import { startCase } from "lodash";
+import Select, { OptionProps, components } from "react-select";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowRight } from "@fortawesome/free-solid-svg-icons";
 import { ToastMsg } from "../notifications/Notifications";
 import GlobalAppContext from "../utils/GlobalAppContext";
+import { FlairEnum, Flair } from "../utils/constants";
+import type { FlairName } from "../utils/constants";
+import Username from "../common/Username";
 
-const flairOptions = [
-  { label: "Default", value: "default", className: "default" },
-  { label: "Fire", value: "fire", className: "fire" },
-  { label: "Water", value: "water", className: "water" },
-  { label: "Air", value: "air", className: "air" },
-  { label: "Disabled", value: "disabled", className: "disabled" },
-];
+function CustomOption(
+  props: OptionProps<{ value: Flair; label: FlairName; username: string }>
+) {
+  const { label, data } = props;
+  return (
+    <components.Option {...props}>
+      <div className="flex" style={{ gap: "1em" }}>
+        <span>{label}</span>
+        <span style={{ flex: 0.5 }}>
+          <FontAwesomeIcon icon={faArrowRight} />
+        </span>
+        <Username
+          style={{ textAlign: "right" }}
+          username={data.username}
+          selectedFlair={data.value}
+          hideLink
+          elementType="a"
+        />
+      </div>
+    </components.Option>
+  );
```

Comment:
We'll probably have to adjust the flex css such that
1. The Underline is limited to the text
2. The Username for the sliced text is not in the middle, instead it should be right alligned.
![image](https://github.com/user-attachments/assets/d09cd98a-ef67-40f7-843e-59cf647d4393)